src/python/pants/help/help_printer.py

```diff
@@ -133,13 +139,13 @@ def _print_options_help(self):
 
       print(self._format_help(ScopeInfo(GLOBAL_SCOPE, ScopeInfo.GLOBAL)))
 
-  def _format_help(self, scope_info):
-    """Return a help message for the options registered on this object.
-
-    Assumes that self._help_request is an instance of OptionsHelp.
-
-    :param scope_info: Scope of the options.
-    """
+  def _format_help(self, scope_info: ScopeInfo) -> str:
+    """Return a help message for the options registered on this object."""
+    if not isinstance(self._help_request, OptionsHelp):
+      raise AssertionError(
+        f"`self._help_request` was expected to have type OptionsHelp but had type "
+        f"{type(self._help_request)}"
+      )
```

Comment:
We assumed this previously - now, to make MyPy happy + defensive programming, we check that the assumption holds.